src/vivarium/component.py

```diff
@@ -627,10 +630,14 @@ def build_lookup_table(
         return builder.lookup.build_table(data)
 
     def _get_columns(
-        self, value_columns: Sequence[str] | None, data: float | pd.DataFrame
-    ) -> tuple[list[str], list[str], list[str]]:
+        self, value_columns: Sequence[str] | None, data: pd.DataFrame
+    ) -> tuple[Sequence[str], list[str], list[str]]:
         all_columns = list(data.columns)
         if value_columns is None:
+            if not self.get_value_columns:
```

Comment:
`self.get_value_columns` will never actually be `None` at this point, since it's being set earlier in `setup_component()`. I think it's better to ignore the mypy error than introduce this check that will never trigger.

Reply:
I can cast?